### PR TITLE
Address a corner case when establishing p2p access

### PR DESCRIPTION
**Pull request by findmyway**
This PR is mainly to fix the issue I reported here: https://github.com/JuliaGPU/NCCL.jl/issues/58

I think a better approach may be adding locks to `peer_access`?

### CODE REVIEW

File: lib/cudadrv/context.jl [hunk 1]
```diff
@@ -402,8 +402,13 @@ function maybe_enable_peer_access(src::CuDevice, dst::CuDevice)
                     enable_peer_access(context(dst))
                     peer_access[][src_idx, dst_idx] = 1
                 catch err
-                    @warn "Enabling peer-to-peer access between $src and $dst failed; please file an issue." exception=(err,catch_backtrace())
-                    peer_access[][src_idx, dst_idx] = -1
+                    if err.code == CUDA.CUDA_ERROR_PEER_ACCESS_ALREADY_ENABLED
+                        @warn "peer-to-peer access between $src and $dst is already enabled"
```
Comment (maleadt) [inline lib/cudadrv/context.jl]:
```suggestion
                        @warn "Peer-to-peer access between $src and $dst was unexpectedly already enabled"
```

**Comment by maleadt** [comment]
I take it you were running with multiple threads? A lock would be useful then, but it can't hurt handling this corner case gracefully too.

**Comment by findmyway** [comment]
> I take it you were running with multiple threads?

No, I didn't run with multiple threads.

<hr>

I tried twice and with the fix here I can see two warnings printed.

**Comment by maleadt** [comment]
Huh, I'm not entirely sure how this happens then. In any case, thanks for the fix!

**Pull request merged**